packages/attune/src/api/error.rs

```diff
@@ -0,0 +1,45 @@
+use axum::{
+    body::Body,
+    http::StatusCode,
+    response::{IntoResponse, Response},
+};
+use bon::Builder;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Builder, Debug)]
+pub struct ErrorResponse {
+    /// The HTTP status code.
+    #[serde(skip)]
+    status: StatusCode,
+    /// A short, unique error code.
+    #[builder(into)]
+    pub error: String,
+    /// A human-readable error message.
+    #[builder(into)]
+    pub message: String,
+}
+
+impl ErrorResponse {
+    pub fn new(status: StatusCode, error: String, message: String) -> Self {
+        Self {
+            status,
+            error,
+            message,
+        }
+    }
+
+    pub fn not_found(entity: &str) -> Self {
+        Self {
+            status: StatusCode::NOT_FOUND,
+            error: format!("{}_NOT_FOUND", entity.to_uppercase()),
+            message: format!("{} not found", entity),
+        }
+    }
```

Comment:
[optional] one way to avoid having to create full `String`s to pass in, while still retaining flexibility, is to either:
- use the builder (to replace `new`)
- accept `entity: impl AsRef<str>` for `not_found`

Reply:
Yes, I'm saying "rather than having to choose between `String` and `&str`, you can just do this to accept both"

Reply:
Let's say I take an argument `s: T` where `T: AsRef<str>`. In that case, all I can do with `s` is create `&str` via `s.as_ref()`, right? But if I need a `String` to begin with, doesn't that mean I now _must_ do an allocation, whereas previously I could possibly be moving in an allocation from the caller?

My current thinking is to take `String` when I need one anyway and `&str` otherwise.

Reply:
If your function needs `String`: accept `impl Into<String>`
If your function needs `&str`: accept `impl AsRef<str>`

Then you perform minimal allocations, based only on what your function actually needs; you also have an much easier to use API